src/classes/OPP_OpportunityContactRoles_TDTM.cls

```diff
@@ -112,11 +112,12 @@ public without sharing class OPP_OpportunityContactRoles_TDTM extends TDTM_Runna
         }
         
         Map<Id,Account> primaryAccounts = new Map<Id,Account>([select id, npe01__One2OneContact__c, npe01__SYSTEM_AccountType__c from Account where id IN :oppAccounts]);
-        List<Contact> primaryContacts = [select AccountId,Account.Id,Account.npe01__SYSTEM_AccountType__c from Contact where Id IN :primaryContactIds];
-
-        for (Contact thisContact : primaryContacts)
-            if (thisContact.AccountId!=null)
-                contactsAndOneToOneAccounts.put(thisContact.Id,thisContact.Account);
+        if (!primaryContactIds.isEmpty()) {
+            for (Contact thisContact : [select AccountId,Account.Id,Account.npe01__SYSTEM_AccountType__c from Contact
+                    where Id IN :primaryContactIds AND AccountId != null]) {
```

Comment:
I believe that "indexes that do not include nulls" means that the Index itself has been updated/recreated to include nulls in it, since by default all Indexes do not include null values. An admin has to specifically request to include nulls in Indexes from Salesforce support.